src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -140,6 +150,115 @@ private List<PluginFile> GetPluginFiles(CancellationToken cancellationToken)
             return files;
         }
 
+        /// <summary>
+        /// Gets auth plugins installed using dotnet tools. This is done by iterating through each file in directories found int eh
+        /// `PATH` environment variable.
+        /// The files are also expected to have a name that starts with `nuget-plugin-`
+        /// </summary>
+        /// <returns></returns>
+        private List<PluginFile> GetNetToolsPluginFiles()
+        {
+            var pluginFiles = new List<PluginFile>();
+
+            if (_environmentVariableReader is null)
+            {
+                _environmentVariableReader = EnvironmentVariableWrapper.Instance;
+            }
```

Comment:
I believe if we change the `_environmentVariableReader` variable to read-only, we can skip this `if` block. In the public constructor, we can add `_environmentVariableReader = EnvironmentVariableWrapper.Instance;`, while in the internal constructor added in this PR, it can be passed as a parameter.

Reply:
we should minimize constructors that don't have empty bodies, to reduce risk of bugs. Look at this one as an example: https://github.com/NuGet/NuGet.Client/blob/785d8bb948a8b477e9160f6732e7df50a1bba5d1/src/NuGet.Clients/NuGet.VisualStudio.Common/Services/NuGetFeatureFlagService.cs#L24-L36

The public constructor just calls the internal constructor with static values, and all the code to initialize the class is in that one internal that is testable with mocks. Importantly, if changes are needed to this class in the future, there's no question where changes to a constructor happen, because there's only one where "real" stuff is happening vs needing to choose from multiple constructors and then some code paths have an unassigned variable.